### PR TITLE
Adds ClientForMetadataTokenRequestEmptyAud and EmptyAudienceTokenFromMetadata

### PR DESCRIPTION
**Pull request by JoshVanL**
This PR adds the `manager/util` package which is intended to include helper functions for csi-lib consumers.

`EmptyAudienceTokenFromMetadata` will return the empty audience request token from the volume context. This is useful for drivers who want to extract the Pod's ServiceAccount information without needed to GET Pods.

`ClientForMetadataTokenRequestEmptyAud` is intended to be plugged straight into `manager.ClientForMetadata` and will be used to generate cert-manager clients who authenticate to the API server using the Pod's ServiceAccount. 

I have ended up copying these funcs multiple times, so seems sensible to add these util helpers to csi-lib so other consumers don't have to write them themselves.

```release-note
NONE
```

/assign @munnerz

### CODE REVIEW

File: manager/util/tokenrequest.go
Comment (irbekrm) [inline manager/util/tokenrequest.go:33]: Suggestion: I understand that the empty audience is so that the audience defaults to kube API server. Would it make sense to add that to the comment or is it unlikely that a user of the function would not already know that?

File: manager/util/tokenrequest.go
Comment (irbekrm) [inline manager/util/tokenrequest.go:51]: And I guess we validate those in `cmclient.NewForConfig` 👍🏼

File: manager/util/tokenrequest.go
Comment (irbekrm) [inline manager/util/tokenrequest.go:53]: Suggestion: indicate in the function name that it's cert-manager client.

**Review by irbekrm — APPROVED** [review]
Looks good to me from what I can tell - I have read through the code and tried to understand the context.

I guess once this gets merged, the functions would be used in `cert-manager-csi`? Then it might make sense to come back and add a link to where they are used.

I've added two optional suggestions, but feel free to unhold.
/hold
/lgtm

File: manager/util/tokenrequest.go
Comment (JoshVanL) [reply manager/util/tokenrequest.go:33]: I've tried to add this context in the following sentence:

> The resulting cert-manager client is
// authenticated against the Kubernetes API server using the mounting Pod's
// ServiceAccount.

It is not that it _defaults_ to the kube API server, it _is_ the kube API server audience, [see](https://kubernetes-csi.github.io/docs/token-requests.html#usage).

File: manager/util/tokenrequest.go
Comment (JoshVanL) [reply manager/util/tokenrequest.go:51]: We don't actually, but we can't really.. As a consumer, you get what you have given.

File: manager/util/tokenrequest.go
Comment (JoshVanL) [reply manager/util/tokenrequest.go:53]: I'm inclined to say no. All consumer facing clients are cert-manager clients, [see](https://github.com/cert-manager/csi-lib/blob/dd8ae5d66f5301940c865704f3e91992ab3bc4d6/manager/manager.go#L58).

I think extra context would be needed if it _wasn't_ a cert-manager client.

**Comment by JoshVanL** [comment]
Thanks for the review @irbekrm :pray:

> I guess once this gets merged, the functions would be used in cert-manager-csi? Then it might make sense to come back and add a link to where they are used.

Not sure if that's needed too much. Ideally this will be used in multiple projects, I have a couple I need it for :smile:

File: manager/util/tokenrequest.go
Comment (irbekrm) [reply manager/util/tokenrequest.go:53]: Ah ok, that makes sense. I was thinking about it like about a controller where the default client think is a core kube client 👍🏼

File: manager/util/tokenrequest.go
Comment (irbekrm) [reply manager/util/tokenrequest.go:33]: Yeah makes sense especially since it's in the CSI dev docs too which I didn't realize 👍🏼

**Comment by irbekrm** [comment]
Thanks for replying to all the comments!

/hold cancel

**Pull request merged**
